src/Admin/Polyfills.php

```diff
@@ -35,26 +32,6 @@ public static function get_registration_action() {
 		return 'amp_register_polyfills';
 	}
 
-	/**
-	 * Check whether the conditional object is currently needed.
-	 *
-	 * @return bool Whether the conditional object is needed.
-	 */
-	public static function is_needed() {
-		return ! Services::get( 'dependency_support' )->has_support();
```

Comment:
Re: #7458, I just realized an alternative approach to this which could solve some headaches we've encountered: we could restore this and add the conditions for when the polyfills should be registered. Namely, on the various admin screens. Perhaps that is actually not better since then we'd have to maintain the list of screens here, whereas if we go the other (current) approach we only do the polyfills which the respective service fires the `amp_register_polyfills` action. We just need to make sure that the `amp_register_polyfills` action only ever fires on our dedicated screens.

So yeah, nevermind.

Reply:
With this conditional `amp_register_polyfills` will only get fired on the AMP plugin screens:

https://github.com/ampproject/amp-wp/blob/a931e7a217f56588777f0602974b275032455f30/src/Admin/ValidationCounts.php#L132-L142